Sources/ActorSingletonPlugin/ActorSingletonProxy.swift

```diff
@@ -12,239 +12,387 @@
+//                    if let oldestMessage = self.buffer.take() {
+//                        context.system.deadLetters.tell(DeadLetter(oldestMessage, recipient: context.id))
+//                    }
+//                default:
+//                    context.log.warning("Unable to stash message, error: \(error)", metadata: self.metadata(context))
+//                }
+//            }
+//        }
+//    }
+
+    nonisolated func stop() {
+        Task {
+            try await self.whenLocal { __secretlyKnownToBeLocal in // TODO(distributed): this is annoying, we must track "known to be local" in typesystem instead
+                // TODO: perhaps we can figure out where `to` is next and hand over gracefully?
+                try await __secretlyKnownToBeLocal.handOver(to: nil)
+                __secretlyKnownToBeLocal.manager = nil
             }
         }
     }
 }
 
 // ==== ----------------------------------------------------------------------------------------------------------------
-// MARK: ActorSingletonManager + logging
+// MARK: Logging
 
 extension ActorSingletonProxy {
-    func metadata<Message>(_: _ActorContext<Message>) -> Logger.Metadata {
+    func metadata() -> Logger.Metadata {
         var metadata: Logger.Metadata = [
             "tag": "singleton",
             "singleton/name": "\(self.settings.name)",
-            "singleton/buffer": "\(self.buffer.count)/\(self.settings.bufferCapacity)",
+            "singleton/buffer": "\(self.remoteCallContinuations.count)/\(self.settings.bufferCapacity)",
         ]
 
         metadata["targetNode"] = "\(optional: self.targetNode?.debugDescription)"
-        if let ref = self.ref {
-            metadata["ref"] = "\(ref.id)"
+        if let singleton = self.singleton {
+            metadata["singleton"] = "\(singleton.id)"
         }
-        if let managerRef = self.managerRef {
-            metadata["managerRef"] = "\(managerRef.id)"
+        if let manager = self.manager {
+            metadata["manager"] = "\(manager.id)"
         }
 
         return metadata
     }
 }
 
 // ==== ----------------------------------------------------------------------------------------------------------------
-// MARK: Singleton path / address
+// MARK: Actor ID and path
 
+// FIXME: remove _system in path?
 extension ActorID {
-    static func _singletonProxy(name: String, remote node: UniqueNode) -> ActorID {
-        ._make(remote: node, path: ._singletonProxy(name: name), incarnation: .wellKnown)
+    static func _singleton(name: String, remote node: UniqueNode) -> ActorID {
+        ._make(remote: node, path: ._singleton(name: name), incarnation: .wellKnown)
     }
 }
 
 extension ActorPath {
-    static func _singletonProxy(name: String) -> ActorPath {
-        try! ActorPath._system.appending("singletonProxy-\(name)")
+    static func _singleton(name: String) -> ActorPath {
+        try! ActorPath._system.appending("singleton-\(name)")
```

Comment:
ok for now, we should move away from paths entirely though -- there's a new constructor on path which does not take a path and we should use tags to mark it as "this is a singleton" etc...

Dont have to do this in this PR tho, just FYI on direction

Reply:
I reworked this bit in https://github.com/apple/swift-distributed-actors/pull/980/commits/1de268f71ab7fca8c36fec52a3a557d5c18155d0

I guess only the actual singleton DA needs to have "known name". `ClusterSingleton` (previously `ActorSingletonProxy`) and `ClusterSingletonBoss` (`ActorSingletonManager`) don't need to be looked up, so they can be named whatever. The singleton DA though, needs to be well-known so we can construct `ActorID` for it. For now the code still uses `knownActorName` and `path`, but maybe we want to do things differently now that we have `ActorMetadata`.

Reply:
I'll dig into this -- the "known" won't work at all once we remove paths since the "known" is just "id == 0" and we cannot have more of those -- previously we could since in the akka and path model the identity is "path + id" but now it is only id...

SO! This is why the metadata was so important, now we can have the metadata participate in identity and this way we can still do "singleton is well known" with a special tag about it. I'll figure this out today into this PR 👍